music21/opensheetmusicdisplay/osmd.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#-------------------------------------------------------------------------------
+# Name:         opensheetmusicdisplay/osmd.py
+# Purpose:      music21 classes for displaying music21 scores in IPython notebooks
+#
+# Authors:      Sven Hollowell
+#
+# Copyright:    Copyright © 2012-14 Michael Scott Cuthbert and the music21 Project
+# License:      LGPL or BSD, see license.txt
+#-------------------------------------------------------------------------------
+'''
+Registers the .show('osmd') converter for use in IPython browser
+See OSMD project page here: https://github.com/opensheetmusicdisplay/opensheetmusicdisplay
+'''
+import unittest
+from music21 import exceptions21
+
+
+class OpenSheetMusicDisplayException(exceptions21.Music21Exception):
+    pass
+
+hasInstalledIPython = False
+try:
+    # raise ImportError('for testing')
+    from IPython.core.display import display, HTML, Javascript
+    hasInstalledIPython = True
+except ImportError as e:
+    # raise error only if methods are called by a .show()
+    def display(*args, **kwargs):
+        raise OpenSheetMusicDisplayException('OpenSheetMusicDisplay requires IPython to be installed')
+    HTML = Javascript = display
+
+
+from music21.converter.subConverters import SubConverter
+from music21.instrument import Piano
+from urllib.request import pathname2url
+import time, random, json, os
+
+
+class ConverterOpenSheetMusicDisplay(SubConverter):
+    '''
+    Converter for displaying sheet music inline
+    in an IPython notebook.
+    Uses: https://opensheetmusicdisplay.org/ for rendering
+    '''
+    registerFormats = ('osmd',)
+    registerShowFormats = ('osmd',) 
+
+    def show(self, obj, fmt, 
+            fixPartName=True, offline=True, div_id=None,
+            **keywords):
+        score = obj
```

Comment:
Does obj need to be a stream.Score()?  can it be any stream, a part, a measure, a single note, etc.?  Generally, we allow .show() to be run on any musicxml element, so if it's not a supported element, we wrap it so it can be show.

This might be a place where the code to do all this in musicxml should be factored out into the generic subconverter class so it can be used in any converter method.